test/integration/fake_upstream.cc

```diff
@@ -344,6 +344,23 @@ void FakeHttpConnection::onGoAway(Http::GoAwayErrorCode code) {
   ENVOY_LOG(info, "FakeHttpConnection receives GOAWAY: ", code);
 }
 
+void FakeHttpConnection::raiseGoAway() {
```

Comment:
I'm pretty sure both of these functions need to use `executeOnDispatcher`.

Also, I would call them `sendGoAway` and `sendProtocolError` or `encodeGoAway`, etc. Whichever matches the other functions.